tasks/spi/src/main/kotlin/Task.kt

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (C) 2024 The ORT Server Authors (See <https://github.com/eclipse-apoapsis/ort-server/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.eclipse.apoapsis.ortserver.tasks.spi
+
+/**
+ * An interface describing a task that can be executed by an external scheduler implementation.
+ */
+interface Task {
+    /**
+     * Execute the specific logic of this task.
+     */
+    suspend fun execute()
```

Comment:
I have added documentation that tasks can throw arbitrary exceptions. Using a `Result` with a generic type as return value seems a bit strange to me, as it would imply that such results could be processed somehow in a generic way.